File: src/Composition/NoteExtender.java
package Composition;

import Finals.NoteDurations;
import java.util.Random;

public class NoteExtender {
    public static String[] Extend(int size, int minDuration, int maxDuration, String[] notes, int staffQuantity) {
        String[] duration = NoteDurations.durations();
        Random durationSelector = new Random();

        String[] staffs = new String[staffQuantity];
        Random staffSelector = new Random();

        Random slurs = new Random();
        int firstRest = staffQuantity - 1;

        for (int j = 0 ; j < staffs.length ; j ++) {
            staffs[j] = "";
        }

        int noteCounter = 0;

        for (int i = 0 ; i <= size ; i++) {
            String slurSymbol = "~";
            String actualNote = notes[noteCounter];
            int actualStaff = staffSelector.nextInt(staffQuantity);
            int quantitySlurs = slurs.nextInt(4)+1;

            if (firstRest != 0) {

                if (staffs[actualStaff] == "" || firstRest != staffQuantity -1) {
                    for (int slur = 0 ; slur <= quantitySlurs ; slur++) {
                        staffs[actualStaff] += "r";
                        staffs[actualStaff] += duration[durationSelector.nextInt(maxDuration)] + " ";
                    }

                    for (int slur = 0 ; slur <= quantitySlurs ; slur++) {
                        staffs[actualStaff] += actualNote;
                        if (slur == quantitySlurs) {
                            slurSymbol = "";
                        }
                        staffs[actualStaff] += duration[durationSelector.nextInt(maxDuration)] + slurSymbol + " ";
                    }
                }

                firstRest -= 1;

                if (noteCounter == notes.length-1) {
                    noteCounter = 0;
                } else {
                    noteCounter += 1;
                }

                continue;
            }

            for (int slur = 0 ; slur <= quantitySlurs ; slur++) {
                staffs[actualStaff] += actualNote;
                if (slur == quantitySlurs) {
                    slurSymbol = "";
                }
                staffs[actualStaff] += duration[durationSelector.nextInt(maxDuration)] + slurSymbol + " ";
            }

            if (noteCounter == notes.length-1) {
                noteCounter = 0;
            } else {
                noteCounter += 1;
            }

        }

        return staffs;
    }
}

File: src/Composition/NoteExtender.kt
package Composition

import Finals.NoteDuration
import Finals.listOfNoteDurations
import java.util.*

fun extendNotes(size: Int, staffQuantity: Int, listOfNotes: List<String>): MutableList<Staff> {
    val durations = listOfNoteDurations()
    val staffs = mutableListOf<Staff>()

    var noteIndex = 0

    repeat(staffQuantity) {
        staffs.add(Staff())
    }

    for (index in 0..size) {
        val selectedStaff = (0..(staffQuantity-1)).random()
        val quantitySlurs = (0..4).random()

        if (staffs[selectedStaff].note.size == 0) {
            for (i in 0..quantitySlurs) {
                val note = "r"
                val duration = durations[(0..8).random()] // static max/min durations
                val symbol = ""

                staffs[selectedStaff].note.add(note)
                staffs[selectedStaff].duration.add(duration)
                staffs[selectedStaff].symbol.add(symbol)
            }

            continue
        }

        for (i in 0..quantitySlurs) {
            val note = listOfNotes[noteIndex]
            val duration = durations[(0..8).random()] // static max/min durations
            val symbol = ""

            staffs[selectedStaff].note.add(note)
            staffs[selectedStaff].duration.add(duration)
            staffs[selectedStaff].symbol.add(symbol)
        }

        if (noteIndex == listOfNotes.size -1) {
            noteIndex = 0
        } else {
            noteIndex += 1
        }
    }

    return staffs
}

data class Staff (
        val note:  MutableList<String> = ArrayList(),
        val duration: MutableList<NoteDuration> = ArrayList(),
        val symbol: MutableList<String> = ArrayList()
)